src/init.cpp

```diff
@@ -1353,6 +1342,10 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     RegisterNodeSignals(GetNodeSignals());
 
+    assert(!connman);
+    connman = std::make_unique<CConnman>();
+    CConnman::Options connOptions;
```

Comment:
Maybe it's just a feeling, but it seems a little confusing to me here:

1. we instantiate `connman` with void `connOptions`
2. we prepare `connOptions` (using `connman`)
3. later (at `StartNode`) we reinject `connOptions` in `connman`

I guess it's working fine the current way, nonetheless I'd prefer to have this flow a little bit more fluent and simple (if possible).

Reply:
The description of the sequence of tasks follows the current implementation, and it is derived from `CConnman`  initialization in Bitcoin. I suppose it has been written in this way because AppInit() relies on a few structures and functions of `CConnman` that should be called before starting the actual networking subsystem (with `StartNode()`).

Reply:
Ok makes sense, thanks.